Modules/FindSpinnakerAPI.cmake

```diff
@@ -0,0 +1,131 @@
+  # Windows Spinnaker API path hints
+  SET(SPINNAKER_API_PATH_HINTS
+      "../PLTools/PointGrey/windows/spinnaker"
+    "../../PLTools/PointGrey/windows/spinnaker"
+    "../trunk/PLTools/PointGrey/windows/spinnaker"
+    "$ENV{PROGRAMFILES}/Point Grey Research/Spinnaker"
+    "C:/Program Files/Point Grey Research/Spinnaker"
+    "$ENV{${_x86env}}/Point Grey Research/Spinnaker"
+    "C:/Program Files (x86)/Point Grey Research/Spinnaker"
+    )
+ELSEIF(UNIX)
+  # Ubuntu Spinnaker API path hints
+
+  # get release codename
+  execute_process(COMMAND lsb_release -cs
+    OUTPUT_VARIABLE _RELEASE_CODENAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+  # get bitness suffix
+  IF(CMAKE_SIZEOF_VOID_P EQUAL 4)
+    SET(_SPINNAKER_BITNESS_SUFFIX "win32")
+  ELSE()
+    SET(_SPINNAKER_BITNESS_SUFFIX "x64")
+  ENDIF()
+
+  SET(SPINNAKER_API_PATH_HINTS
+    "../PLTools/PointGrey/ubuntu_${_RELEASE_CODENAME}/spinnaker_${_SPINNAKER_BITNESS_SUFFIX}"
+    "../../PLTools/PointGrey/ubuntu_${_RELEASE_CODENAME}/spinnaker${_SPINNAKER_BITNESS_SUFFIX}"
+    "../trunk/PLTools/PointGrey/ubuntu_${_RELEASE_CODENAME}/spinnaker${_SPINNAKER_BITNESS_SUFFIX}"
+    "/usr/include/spinnaker"
+    )
+ENDIF()
+
+# check for local spinnaker folder
+find_path(SPINNAKER_DIR "/include/System.h"
+  PATHS ${SPINNAKER_API_PATH_HINTS}
+  DOC "Point Grey Spinnaker API directory")
+
+IF (SPINNAKER_DIR)
+  # Include directories
+  set(SPINNAKER_API_INCLUDE_DIR ${SPINNAKER_DIR}/include)
+  mark_as_advanced(SPINNAKER_API_INCLUDE_DIR)
+
+  IF(WIN32)
+    # Windows
+    IF(CMAKE_SIZEOF_VOID_P EQUAL 4)
+      # 32 bits
+      set(SPINNAKER_API_LIBRARY_DIR ${SPINNAKER_DIR}/lib/vs2015)
+      mark_as_advanced(SPINNAKER_API_LIBRARY_DIR)
+      set(SPINNAKER_API_BINARY_DIR ${SPINNAKER_DIR}/bin/vs2015)
+      mark_as_advanced(SPINNAKER_API_BINARY_DIR)
+    ELSE()
+      # 64 bits
+      set(SPINNAKER_API_LIBRARY_DIR ${SPINNAKER_DIR}/lib64/vs2015)
+      mark_as_advanced(SPINNAKER_API_LIBRARY_DIR)
+      set(SPINNAKER_API_BINARY_DIR ${SPINNAKER_DIR}/bin64/vs2015)
+      mark_as_advanced(SPINNAKER_API_BINARY_DIR)
+    ENDIF()
+
+  ELSEIF(UNIX)
+    # Ubuntu
+    set(SPINNAKER_API_LIBRARY_DIR ${SPINNAKER_DIR}/lib)
+    mark_as_advanced(SPINNAKER_API_LIBRARY_DIR)
+    set(SPINNAKER_API_BINARY_DIR ${SPINNAKER_DIR}/lib)
+    mark_as_advanced(SPINNAKER_API_BINARY_DIR)
+  ENDIF()
+
+  #Version
+  spinnaker_get_version("${SPINNAKER_DIR}/include/System.h")
+
+ELSE()
+  # check if spinnaker is globally installed on Ubuntu
+  IF (UNIX AND EXISTS "/usr/include/spinnaker/System.h")
+    set(SPINNAKER_API_INCLUDE_DIR /usr/include/spinnaker)
+    mark_as_advanced(SPINNAKER_API_INCLUDE_DIR)
+    set(SPINNAKER_API_LIBRARY_DIR /usr/lib)
+    mark_as_advanced(SPINNAKER_API_LIBRARY_DIR)
+    set(SPINNAKER_API_BINARY_DIR /usr/lib)
+    mark_as_advanced(SPINNAKER_API_BINARY_DIR)
+
+    #Version
+    spinnaker_get_version("/usr/include/spinnaker/System.h")
+  ENDIF()
+ENDIF()
+
+include(FindPackageHandleStandardArgs)
```

Comment:
Would you consider populating a CMake target with the correct details?

Windows example can be found here:
https://github.com/PlusToolkit/PlusBuild/blob/master/Modules/FindAgilentMD1.cmake#L60
Linux example here:
https://github.com/PlusToolkit/PlusBuild/blob/master/Modules/FindV4L2.cmake#L49